### PR TITLE
Add state alias

### PR DESCRIPTION
**Pull request by Melanitski**
If we have complicated nested route name, we can use `alias`
Example:

```
  .state 'public.search',
    alias : 'search'
    url: 'search'
    template: 'views/search'

# Usage
ui-sref="search"
$state.go("search")
```

### CODE REVIEW

**Comment by eddiemonge** [comment]
Why not:

```
.state 'search',
  parent: 'public'
```
?

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "angular-ui-router",
-  "version": "0.2.15",
+  "version": "0.2.16",
```
Comment (eddiemonge) [inline bower.json]: this happens as part of the build process so should not be in your commit

**Comment by eddiemonge** [comment]
for pull requests like this, only include the actual files related to your change, not the build related things like version changes or updated dist code. Also, one commit per pull request. Also, you should follow the [contributing guide](https://github.com/angular-ui/ui-router/blob/master/CONTRIBUTING.md)

**Comment by Melanitski** [comment]
If we have very complicated router like 'public.search.item.compare' we can make alias like a `compare` or `items-compare` 
We need to add tests and dist files?

**Comment by eddiemonge** [comment]
```
state: 'compare',
  parent: 'public.search.item'
```

**Comment by christopherthielen** [comment]
I agree with @eddiemonge . While I am not fundamentally opposed to an alias, I think we already have a mechanism that provides the desired behavior (`parent: `).  

Also, ui-sref supports relative addressing, so within a template for `public.search.item`, you simply enter a link `<a ui-sref=".compare">Compare</a>`.  Likewise, `$state.go()` also accepts relative state names.

Closing for now; Feel free to respond with additional discussion.

**Pull request closed without merging**